tests/contrib/cherrypy/test_middleware.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+import time
+import re
+
+from ddtrace.contrib.cherrypy import TraceMiddleware
+from ddtrace.constants import SAMPLING_PRIORITY_KEY
+from ddtrace.ext import http, errors
+from ddtrace import config
+
+import cherrypy
+from cherrypy.test import helper
+from .web import TestApp
+from tests.tracer.test_tracer import get_dummy_tracer
+from ... import assert_span_http_status_code
+from six.moves.urllib.parse import quote as url_quote
+import logging
+
+logger = logging.getLogger()
+logger.level = logging.DEBUG
+
```

Comment:
Some additional test cases which would be really nice to have:

- Test a configuring service name (both through configuration and through the middleware directly)
- Test disabling distributed tracing

Reply:
Added tests for configuring service name and distributed trace disabling via both methods. They were initially failing as the tool didn't know whether to use the config object or its own settings. So I have merged the two through the use of properties. Now config.cherrypy is the authoritive store of this configuration, whether it is set in the tool, or through ddtrace.

Reply:
Great solution!